data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ElasticsearchSink.java

```diff
@@ -187,6 +192,10 @@ private void checkAndCreateIndex() throws IOException {
         if (e.getMessage().contains("resource_already_exists_exception")) {
           // Do nothing - likely caused by a race condition where the resource was created
           // by another host before this host's restClient made its request
+        } else if (e.getMessage().contains(String.format(INDEX_ALIAS_USED_AS_INDEX_ERROR, indexAlias))) {
+          throw new ElasticsearchException(
+                  String.format("An index exists with the same name as the reserved index alias name [%s], please delete or migrate the existing index",
```

Comment:
We should consider defining custom Data Prepper exception types instead of re-using ES's (can be another maintenance task).

I'd imagine we would define base DataPrepperException's in the API package, and have plugins define extensions where needed.

Reply:
Great call. We could wrap it into a checked data-prepper exception. For now I think I will wrap it into IOException.